### PR TITLE
Guns For Days

### PR DESCRIPTION
**Pull request by TheDanseMacabre**
Adds a large number of weapons that were previously random gun or admin-spawn only to the uplink. Most are pretty expensive, some prohibitively so.
Also buffs the energy sword, which did paltry damage through armor.

Some of these weapons may be too strong for the uplink, or there may be problems with pricing. This PR should probably be test-merged before actual merging, and I expect some things may need to be cut or changed.

### CODE REVIEW

**Comment by TheDanseMacabre** [comment]
!review

File: code/datums/uplink/ammunition.dm [hunk 1]
```diff
@@ -73,4 +73,76 @@
 	name = "K2557 Magazine"
 	item_cost = 2
 	path = /obj/item/ammo_magazine/super_heavy
-	desc = "A spare magazine, for the super heavy K2557 pistol."
\ No newline at end of file
+	desc = "A spare magazine, for the super heavy K2557 pistol."
+
+/datum/uplink_item/item/ammo/shotgun_slug
+	name = "Shotgun Slug"
+	item_cost = 2
```
Comment (SleepyGemmy) [inline code/datums/uplink/ammunition.dm:80]: a single slug for 2 TC?

File: code/datums/uplink/ammunition.dm [hunk 2]
```diff
@@ -73,4 +73,76 @@
...
+	path = /obj/item/ammo_casing/shotgun
+	desc = "A shotgun slug."
+
+/datum/uplink_item/item/ammo/shotgun_buckshot
+	name = "Buckshot Shell"
+	item_cost = 1
```
Comment (SleepyGemmy) [inline code/datums/uplink/ammunition.dm:86]: ditto.

File: code/datums/uplink/ammunition.dm [hunk 3]
```diff
@@ -73,4 +73,76 @@
...
+	path = /obj/item/ammo_magazine/assault_shotgun
+	desc = "A magazine for an assault shotgun, loaded with slug shells."
+
+/datum/uplink_item/item/ammo/buckshot_magazine
+	name = "Buckshot Magazine"
+	item_cost = 4
+	path = /obj/item/ammo_magazine/assault_shotgun/shells
+	desc = "A magazine for an assault shotgun, loaded with buckshot shells."
+
+/datum/uplink_item/item/ammo/buckshot_magazine
```
Comment (SleepyGemmy) [inline code/datums/uplink/ammunition.dm:108]: path.

File: code/datums/uplink/ammunition.dm [hunk 4]
```diff
@@ -73,4 +73,76 @@
...
+	name = "Tungsten Railgun Ammunition"
+	item_cost = 8
+	path = /obj/item/ammo_magazine/trodpack
+	desc = "Two tungsten rods, used by a railgun."
+
+/datum/uplink_item/item/ammo/buckshot_magazine
```
Comment (SleepyGemmy) [inline code/datums/uplink/ammunition.dm:114]: path.

File: code/game/objects/items/weapons/melee/energy.dm [hunk 5]
```diff
@@ -216,12 +216,13 @@
...
 	color
 	name = "energy sword"
-	desc = "May the force be within you."
+	desc = "An energy sword. Quite rare, very dangerous."
 	desc_antag = "The energy sword is a very strong melee weapon, capable of severing limbs easily, if they are targeted.  It can also has a chance \
 	to block projectiles and melee attacks while it is on and being held.  The sword can be toggled on or off by using it in your hand.  While it is off, \
 	it can be concealed in your pocket or bag."
 	icon_state = "sword0"
 	active_force = 30
+	armor_penetration = 25
```
Comment (SleepyGemmy) [inline code/game/objects/items/weapons/melee/energy.dm:225]: note that this is an increase of 5 from the baseline of 20. intentional?

Comment (TheDanseMacabre) [reply code/game/objects/items/weapons/melee/energy.dm:225]: Technically, yes. I wasn't right about the amount of AP the e-sword has atm, but nonetheless 25 is a better value than 20. I want the e-sword to be threatening even to someone in riot/heavy armor, and 25 achieves that without being overwhelming. 20 does substantially less damage.

File: code/datums/uplink/ammunition.dm (hunk 1 above)
Comment (TheDanseMacabre) [reply code/datums/uplink/ammunition.dm:80]: Slugs are pretty strong. Maybe I'll lower it to 1.

File: code/datums/uplink/ammunition.dm (hunk 2 above)
Comment (TheDanseMacabre) [reply code/datums/uplink/ammunition.dm:86]: Going to replace this with a buckshot ammo box later.

File: code/game/objects/items/weapons/melee/energy.dm (hunk 5 above)
Comment (SleepyGemmy) [reply code/game/objects/items/weapons/melee/energy.dm:225]: 👍

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 6]
```diff
@@ -183,3 +183,108 @@
 	name = "Energy Pistol"
 	item_cost = 2
 	path = /obj/item/gun/energy/pistol
+
+/datum/uplink_item/item/visible_weapons/pulse_pistol
+	name = "Pulse Pistol"
+	item_cost = 12
+	path = /obj/item/gun/energy/pulse/pistol
```
Comment (NonQueueingMatt) [inline code/datums/uplink/highly visible and dangerous weapons.dm:190]: This one needs to go.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 7]
```diff
@@ -183,3 +183,108 @@
...
+
+/datum/uplink_item/item/visible_weapons/pulse_carbine
+	name = "Pulse Carbine"
+	item_cost = 16
+	path = /obj/item/gun/energy/pulse
```
Comment (NonQueueingMatt) [inline code/datums/uplink/highly visible and dangerous weapons.dm:195]: Same here.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 8]
```diff
@@ -183,3 +183,108 @@
...
+
+/datum/uplink_item/item/visible_weapons/bullpup_assault_carbine
+	name = "Bullpup Assault Carbine"
+	item_cost = 12
+	path = /obj/item/gun/projectile/automatic/rifle/z8
+
+/datum/uplink_item/item/visible_weapons/bullpup_scout_carbine
+	name = "Bullpup Scout Carbine"
+	item_cost = 20
+	path = /obj/item/gun/projectile/automatic/rifle/w556
```
Comment (NonQueueingMatt) [inline code/datums/uplink/highly visible and dangerous weapons.dm:210]: Not a fan of weapons with scopes being normalised either.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 9]
```diff
@@ -183,3 +183,108 @@
...
+
+/datum/uplink_item/item/visible_weapons/laser_shotgun
+	name = "Laser Shotgun"
+	item_cost = 10
+	path = /obj/item/gun/energy/laser/shotgun
+
+/datum/uplink_item/item/visible_weapons/sniper_rifle
+	name = "Sniper Rifle"
+	item_cost = 16
+	path = /obj/item/gun/projectile/dragunov
```
Comment (NonQueueingMatt) [inline code/datums/uplink/highly visible and dangerous weapons.dm:235]: Similar to above.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

**Comment by Arrow768** [comment]
DNM until we have merged #14999 and the item limits can be incorporated into this PR instead of "prohibitively pricing" things where that makes sense. (Generally that is the case where you want a limited number of a specific gun even across team antags).

File: code/datums/uplink/ammunition.dm (hunk 1 above)
Comment (Zelmana) [reply code/datums/uplink/ammunition.dm:80]: A single slug though? A hacked protolathe can produce them for a few sheets or less of metal. If it is a package, 1 TC would be good. You can only carry so many slugs.

File: code/datums/uplink/ammunition.dm (hunk 2 above)
Comment (Zelmana) [reply code/datums/uplink/ammunition.dm:86]: Slug shot should be more expensive than buckshots, though. Keep slugs at 2 TC.

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 10]
```diff
@@ -183,3 +183,78 @@
...
+	path = /obj/item/gun/energy/rifle/laser/xray
+
+/datum/uplink_item/item/visible_weapons/combat_shotgun
+	name = "Combat Shotgun"
+	item_cost = 10
+	path = /obj/item/gun/projectile/shotgun/pump/combat
+
+/datum/uplink_item/item/visible_weapons/automatic_45_pistol
+	name = "Automatic .45 Pistol"
+	item_cost = 5
```
Comment (Spud5) [inline code/datums/uplink/highly visible and dangerous weapons.dm:224]: The .45 auto submachine gun has similar stats and that costs 8 TC. I think you should raise the price of this or lower the price of the .45 auto SMG.

Comment (TheDanseMacabre) [reply code/datums/uplink/highly visible and dangerous weapons.dm:224]: Capacity for the machine pistol is lower and it can’t use drum magazines

**Review by Arrow768 — APPROVED** [review]

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 11]
```diff
@@ -183,3 +183,78 @@
 	name = "Energy Pistol"
 	item_cost = 2
 	path = /obj/item/gun/energy/pistol
+
+/datum/uplink_item/item/visible_weapons/gyrojet_pistol
```
Comment (Arrow768) [inline code/datums/uplink/highly visible and dangerous weapons.dm:187]: This should be limited with the item limits in addition to the current cost

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 12]
```diff
@@ -183,3 +183,78 @@
...
+	name = "Derringer"
+	item_cost = 1
+	path = /obj/item/gun/projectile/revolver/derringer
+
+/datum/uplink_item/item/visible_weapons/gauss_thumper
+	name = "Gauss Thumper"
+	item_cost = 10
+	path = /obj/item/gun/projectile/gauss
+
+/datum/uplink_item/item/visible_weapons/railgun
```
Comment (Arrow768) [inline code/datums/uplink/highly visible and dangerous weapons.dm:252]: This should be limited with the item limits in addition to the current cost

File: code/datums/uplink/highly visible and dangerous weapons.dm [hunk 13]
```diff
@@ -183,3 +183,78 @@
...
+	name = "Bullpup Assault Carbine"
+	item_cost = 12
+	path = /obj/item/gun/projectile/automatic/rifle/z8
+
+/datum/uplink_item/item/visible_weapons/assault_shotgun
+	name = "Assault Shotgun"
+	item_cost = 15
+	path = /obj/item/gun/projectile/automatic/rifle/shotgun
+
+/datum/uplink_item/item/visible_weapons/gatling_laser
```
Comment (Arrow768) [inline code/datums/uplink/highly visible and dangerous weapons.dm:202]: This should be limited with the item limits in addition to the current cost

**Review by Arrow768 — CHANGES_REQUESTED** [review]
As indicated in the comments

File: code/datums/uplink/ammunition.dm [hunk 14]
```diff
@@ -73,4 +73,70 @@
...
+	name = "7.62 Assault Rifle magazine"
+	item_cost = 2
+	path = /obj/item/ammo_magazine/c762
+	desc = "A magazine for an assault rifle."
+
+/datum/uplink_item/item/ammo/sniper_ammo
+	name = "7.62 Sniper Magazine"
+	item_cost = 4
+	path = /obj/item/ammo_magazine/d762
+	desc = "A magazine for 7.62 sniper rifle."
```
Comment (SleepyGemmy) [inline code/datums/uplink/ammunition.dm:118]:
```suggestion
	desc = "A magazine for a 7.62 sniper rifle."
```

File: code/datums/uplink/ammunition.dm [hunk 15]
```diff
@@ -73,4 +73,70 @@
...
+	name = "20mm Rocket Magazine"
+	item_cost = 4
+	path = /obj/item/ammo_magazine/a75
+	desc = "A 20mm rocket magazine."
+
+/datum/uplink_item/item/ammo/bullpup_magazine
+	name = "5.56 Rifle Magazine"
+	item_cost = 2
+	path = /obj/item/ammo_magazine/a556
+	desc = "A magazine for 5.56 rifle."
```
Comment (SleepyGemmy) [inline code/datums/uplink/ammunition.dm:130]:
```suggestion
	desc = "A magazine for a 5.56 rifle."
```

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**